app/scripts/modules/core/react/react.injector.ts

```diff
@@ -0,0 +1,109 @@
+  public get stateEvents() { return this.$injector.get('stateEvents') as StateEvents; }
+  public get API() { return this.$injector.get('API') as Api; }
+  public get applicationModelBuilder() { return this.$injector.get('applicationModelBuilder') as ApplicationModelBuilder; }
+  public get applicationReader() { return this.$injector.get('applicationReader') as ApplicationReader; }
+  public get authenticationService() { return this.$injector.get('authenticationService') as AuthenticationService; }
+  public get collapsibleSectionStateCache() { return this.$injector.get('collapsibleSectionStateCache') as CollapsibleSectionStateCache; }
+  public get cancelModalService() { return this.$injector.get('cancelModalService') as CancelModalService; }
+  public get cloudProviderRegistry() { return this.$injector.get('cloudProviderRegistry') as CloudProviderRegistry; }
+  public get providerSelectionService() { return this.$injector.get('providerSelectionService') as ProviderSelectionService; }
+  public get clusterFilterService() { return this.$injector.get('clusterFilterService') as ClusterFilterService; }
+  public get confirmationModalService() { return this.$injector.get('confirmationModalService') as ConfirmationModalService; }
+  public get executionFilterModel() { return this.$injector.get('executionFilterModel') as ExecutionFilterModel; }
+  public get executionFilterService() { return this.$injector.get('executionFilterService') as ExecutionFilterService; }
+  public get executionService() { return this.$injector.get('executionService') as ExecutionService; }
+  public get loadBalancerFilterService() { return this.$injector.get('loadBalancerFilterService') as LoadBalancerFilterService; }
+  public get loadBalancerFilterModel() { return this.$injector.get('loadBalancerFilterModel') as LoadBalancerFilterModel; }
+  public get pipelineConfig() { return this.$injector.get('pipelineConfig') as PipelineConfigProvider; }
+  public get pipelineConfigService() { return this.$injector.get('pipelineConfigService') as PipelineConfigService; }
+  public get manualJudgmentService() { return this.$injector.get('manualJudgmentService') as ManualJudgmentService; }
+  public get pipelineTemplateService() { return this.$injector.get('pipelineTemplateService') as PipelineTemplateService; }
+  public get variableInputService() { return this.$injector.get('variableInputService') as VariableInputService; }
+  public get variableValidatorService() { return this.$injector.get('variableValidatorService') as VariableValidatorService; }
+  public get schedulerFactory() { return this.$injector.get('schedulerFactory') as SchedulerFactory; }
+  public get infrastructureSearchService() { return this.$injector.get('infrastructureSearchService') as InfrastructureSearchService; }
+  public get waypointService() { return this.$injector.get('waypointService') as WaypointService; }
+
+  public give(fn: IWantInjector): void {
```

Comment:
That's true, although you've reminded me why I need to deal with the `angular2react` components (didn't do that in this PR but will add them).

Still need the `flush` or something similar to get the `$injector` into the class, but will rename it to `initialize` or something.